tests/components/smappee/test_config_flow.py

```diff
@@ -10,6 +20,105 @@
 CLIENT_SECRET = "5678"
 
 
+async def test_show_user_form(hass):
+    """Test that the user set up form is served."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER},
+    )
+
+    assert result["step_id"] == "user"
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+
+
+async def test_show_zeroconf_confirm_form(hass):
+    """Test that the zeroconf confirmation form is served."""
+    flow = config_flow.SmappeeFlowHandler()
+    flow.hass = hass
+    flow.context = {
+        "source": SOURCE_ZEROCONF,
+        CONF_HOSTNAME: "Smappee1006000212.local.",
+        CONF_TITLE: "Smappee1006000212",
+        CONF_IP_ADDRESS: "1.2.3.4",
+        CONF_SERIALNUMBER: "1006000212",
+    }
+    result = await flow.async_step_zeroconf_confirm()
```

Comment:
We don't need to test this separately since it will be tested in the zeroconf flow test. We should just test each flow from entry to exit point in all permutations.

Reply:
I see. I'm trying to add tests for the zeroconf flow by mocking the HTTP calls. Any pointers to requests based examples (non async)?

Reply:
Patch the library calls with `patch` imported from `tests.async_mock`. Use our scaffold script to generate test example code.

Reply:
> Use our scaffold script to generate test example code.

`python3 -m script.scaffold --integration smappee config_flow` doesn't generate test example code?

Reply:
Here's the template:
https://github.com/home-assistant/core/blob/dev/script/scaffold/templates/config_flow/tests/test_config_flow.py

Reply:
Thanks for the patch hint